File: ApiProject/API-TFG/src/main/java/com/aitor/api_tfg/user/UserResponse.java
package com.aitor.api_tfg.user;

import com.aitor.api_tfg.model.user.Gender;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserResponse {
    private Integer id;
    private String Username;
    private String nombre;
    private String apellidos;
    private String email;
    private Double peso;
    private Integer altura;
    private Gender genero;
}

File: AndroidProject/app/src/main/java/com/aitor/trackactividades/authentication/data/response/UserResponse.kt
package com.aitor.trackactividades.authentication.data.response

import com.aitor.trackactividades.core.model.Gender
import kotlinx.serialization.Serializable
import java.time.LocalDate

@Serializable
data class UserResponse(
    val id: Int,
    val username: String,
    val nombre: String,
    val apellidos: String,
    val email: String,
    val peso: Double,
    val altura: Int,
    val genero: Gender
) {
    override fun toString(): String {
        return """
            UserResponse(
                id=$id,
                username='$username',
                nombre='$nombre',
                apellidos='$apellidos',
                email='$email',
                peso=$peso,
                altura=$altura,
                genero='$genero'
            )
        """.trimIndent()
    }
}

